File: LeetCode/Easy/0383-ransom-note/0383-ransom-note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        freqDict = {}
        for char in magazine:
            if char in freqDict:
                freqDict[char]+=1
            else:
                freqDict[char] = 1
        

        for char in ransomNote:
            if char in freqDict and freqDict[char]>0:
                freqDict[char]-=1
            else:
                return False

        return True
        

File: LeetCode/Easy/0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char, int> mag;
        for (char c : magazine) {
            mag[c]++;
        }
        for (int i = 0; i < ransomNote.size(); i++) {
            if (mag[ransomNote[i]] <= 0) {
                return false;
            }
            mag[ransomNote[i]]--;
        }
        return true;
    }
};